python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -175,7 +175,25 @@ def have_all_requested_runs_finished(self) -> bool:
 
         return True
 
-    def get_target_root_asset_partitions(self) -> Iterable[AssetKeyPartitionKey]:
+    def get_target_root_asset_partitions(
+        self, instance_queryer: CachingInstanceQueryer
+    ) -> Iterable[AssetKeyPartitionKey]:
+        def _get_self_and_downstream_targeted_subset(
```

Comment:
is there a way to get a before/after measurement of the performance of this on a larger graph w/ lots of partitions? This example: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster-test/dagster_test/toys/auto_materializing/large_graph.py might be a good potential one to try it out on.

My concern here is that `bfs_filter_asset_partitions` requires you to individually partition map every single partition key in the subset, which can take a very long time compared to just mapping from a large set of keys to the large set of downstream keys.

I was thinking we could instead just use `bfs_filter_subsets` with a condition function of "this asset key is one of the asset keys targeted by the backfill". This would potentially end up with a subset that contains more partitions than we want (as it's not guaranteed that all children partitions will be part of the target subset, even if some partitions for that key are), but we could filter those extra partitions out after the fact by just taking the intersection between the target subset and the subset we get from the process described above. That might require adding another operation to the asset graph subset similar to: https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/definitions/asset_graph_subset.py?L142, or whatever makes sense to you.

My expectation is that the current implementation would take quite awhile on the example repo, but the bfs_filter_subsets would be significantly faster (open to being proven wrong on either count there)

Reply:
Thanks for catching this, your concerns about perf are very justified given that the original implementation took over 5 minutes to run the function on the large asset graph 😅 

I switched to `bfs_filter_subsets` -- thanks for the pointer, didn't know this function existed. On the large asset graph it takes less than 15 seconds, which I think is ok given that this is a one-time cost per-backfill and backfills are typically long running anyway